File: 3805-maximize-active-section-with-trade-i/maximize-active-section-with-trade-i.java
class Solution {
    public int maxActiveSectionsAfterTrade(String s) {
        int totalOnes = 0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='1'){
                totalOnes++;
            }
        }

        String t = "1"+s+"1";
        List<Integer> arr = new ArrayList<>();
        int n = t.length();
        int i=0;
        while(i<n){
            if(t.charAt(i)=='1'){
                while(i<n && t.charAt(i)=='1'){
                    i++;
                }
            }else{
                int start = i;
                while(i<n && t.charAt(i)=='0'){
                    i++;
                }
                arr.add(i-start);
            }
        }

        int max = 0;
        for(int k=0;k<arr.size()-1;k++){
            int gain = arr.get(k)+arr.get(k+1);
            max = Math.max(max,gain);
        }
        return max+totalOnes;
    }
}

File: 3805-maximize-active-section-with-trade-i/maximize-active-section-with-trade-i.py
class Solution:
    def maxActiveSectionsAfterTrade(self, s: str) -> int:
        totalOnes = 0
        for i in s:
            if i == '1':
                totalOnes += 1
        
        t = '1'+s+'1'
        arr = []
        i = 0
        n = len(t)
        while i<n:
            if t[i] == '1':
                while i<n and t[i] == '1':
                    i+=1
            else:
                start = i
                while i<n and t[i] == '0':
                    i+=1
                arr.append(i-start)
        m = 0
        for i in range(len(arr)-1):
            gain = arr[i]+arr[i+1]
            m = max(m,gain)
        return m+totalOnes